src/Microsoft.AspNet.Authentication/ClaimsTransformationAuthenticationHandler.cs

```diff
@@ -0,0 +1,106 @@
+// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
```

Comment:
What is the code the user would have to write to set it up?
All we seem to be doing is exposing a Delegate that performs claims transformation.  ClaimsPrincipal in, ClaimsPrincipal out.

Why does that require a whole MiddleWare?  It has to be maintained, tested, etc.
Where would the middleware go in the layering of middlewares?